src/box/gc.c

```diff
@@ -744,6 +822,411 @@ gc_consumer_update(struct gc_consumer *consumer, const struct vclock *vclock)
 		gc_schedule_cleanup();
 }
 
+void
+gc_consumer_update(struct gc_consumer *consumer, const struct vclock *vclock)
+{
+	if (consumer->is_persistent) {
+		vclock_copy(&consumer->volatile_vclock, vclock);
+		/*
+		 * Do not wake up the background fiber if it's persisting
+		 * consumers right now because it may be waiting for a cbus
+		 * message, which doesn't tolerate spurious wakeups.
+		 */
+		if (gc.persist_scheduled++ == gc.persist_completed)
+			fiber_wakeup(gc.persist_fiber);
+	} else {
+		gc_consumer_update_impl(consumer, vclock);
```

Comment:
Why a separate path for persistent consumers? Let's update the in-memory state and schedule background sync if the consumer is persistent.